ui/pages/destinations/index.js

```diff
@@ -0,0 +1,201 @@
+
+            const users = new Set(grants?.filter(g => !g?.subject?.startsWith('g:'))).size
+            const groups = new Set(grants?.filter(g => g?.subject?.startsWith('g:'))).size
+
+            return (
+              <div className='flex text-right justify-end w-24 h-8 ml-auto'>
+                {grants && (
+                  <div className='group-hover:hidden flex justify-center items-center text-gray-300'>
+                    {users === 0 && groups === 0
+                      ? (
+                        <div>
+                          No access
+                        </div>
+                        )
+                      : (
+                        <>
+                          {users > 0 && (
+                            <div>
+                              {users}&nbsp;User{users > 1 && 's'}
+                            </div>
+                          )}
+                          {users > 0 && groups > 0 && (
+                            <div className='mx-1'>•</div>
+                          )}
+                          {groups > 0 && (
+                            <div>
+                              {groups}&nbsp;Group{groups > 1 && 's'}
+                            </div>
+                          )}
+                        </>
+                        )}
+                  </div>
+                )}
+                <div className='group-hover:flex space-x-1 hidden'>
+                  <button onClick={() => setShareOpen(true)} className='cursor-pointer bg-zinc-900 rounded-lg'>
+                    <div className='flex items-center py-1 px-3 text-gray-500 hover:text-white'>
+                      <ShareIcon className='w-4 h-4 ' /><div className='text-sm ml-1'>Grant</div>
+                    </div>
+                  </button>
+
+                  {/* grant modal */}
+                  <Grant
+                    id={row.original.id}
+                    modalOpen={shareOpen}
+                    handleCloseModal={() => setShareOpen(false)}
+                  />
+                </div>
+              </div>
+            )
+          }
+        }]
+      : [],
+    ...admin
+      ? [{
+          id: 'remove',
+          accessor: d => d,
+          Cell: ({ rows, value: { id, name } }) => {
+            const { mutate } = useSWRConfig()
+
+            const [open, setOpen] = useState(false)
+
+            return (
+              <div className='flex justify-end w-6 ml-auto opacity-0 group-hover:opacity-100'>
+                <button onClick={() => setOpen(true)} className='py-1 px-2 -mr-2 cursor-pointer'>
+                  <XIcon className='w-5 h-5 text-gray-500 hover:text-white' />
+                </button>
+
+                {/* delete modal */}
+                <DeleteModal
+                  open={open}
+                  setOpen={setOpen}
+                  onSubmit={async () => {
+                    await fetch(`/v1/destinations/${id}`, { method: 'DELETE' })
```

Comment:
☝🏻 is this here by accident? why are we calling delete endpoint twice?

Reply:
Great catch. Fixed.